module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,59 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+CompressedBitmap compressGrayscale(const Bitmap& bitmap) {
+    CompressedBitmap compressed;
+    compressed.reserve(width * height);
+
+    int row{};
+    do {
+        for (auto beg = bitmap[row].begin(),
+                  end = bitmap[row].end();
+             beg != end;) {
+            const auto diff = std::find_if(beg, end,
+                                           [&beg](const auto& el) {
+                                               return el != *beg;
+                                           });
+            compressed.emplace_back(*beg, std::distance(beg, diff));
+            beg = diff;
+        }
+    } while (++row != height);
+
+    compressed.shrink_to_fit();
+    return compressed;
+}
+
+Bitmap decompressGrayscale(const CompressedBitmap& compressed) {
+    Bitmap bitmap;
+
+    size_t count{}, line{}, el{};
```

Comment:
I know we waste space, but please one declaration/ definition per line :)

Reply:
I understand; easier to read :) Thanks